compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -986,6 +985,27 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                 num_supplied_defaults += 1;
             }
         }
+
+        let mut const_params = generics
```

Comment:
It would be better to merge these two loops, for types and consts.

Reply:
If I join them, I have to carry the kind of GenericParamDefKind through, I think it's more efficient to run (maybe not to read), than joining the two loops. If you feel strongly about it, I can change it, but I think as is is fine.

Reply:
I doubt there's much difference in efficiency: after all, with the loops merged you need fewer iterations. But I think the increase in code readability is worth the change.